### PR TITLE
Refactor AES key generation

### PR DESCRIPTION
**Pull request by ivanyu**
We're going to replace Bouncy Castle with the JDK crypto library, which cannot generate 512 bit AES keys. We don't in fact need 512 bits, it was just used to simultaneously generate a key and an AAD. Effectively, splitting in two calls in this commit.

### CODE REVIEW

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java [hunk 1]
```diff
@@ -30,37 +28,24 @@
...
-
     public DataKeyAndAAD createDataKeyAndAAD() {
-        final byte[] dataKeyAndAAD = createDataKey().getEncoded();
-        final byte[] dataKey = new byte[KEY_AND_AAD_SIZE_BYTES];
-        System.arraycopy(dataKeyAndAAD, 0, dataKey, 0, 32);
-        final byte[] aad = new byte[KEY_AND_AAD_SIZE_BYTES];
-        System.arraycopy(dataKeyAndAAD, 32, aad, 0, 32);
-        return new DataKeyAndAAD(new SecretKeySpec(dataKey, "AES"), aad);
+        final var dataKey = aesKeyGenerator.generateKey();
+        final byte[] aad = aesKeyGenerator.generateKey().getEncoded();
```
Comment (jeqo) [inline commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:47]: Should reuse the data key?
```suggestion
        final var dataKey = aesKeyGenerator.generateKey();
        final byte[] aad = dataKey.getEncoded();
```

Comment (ivanyu) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:47]: No, these two are separate things, they must be different

Comment (ivanyu) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java:49]: Made a note and test for this

File: commons/src/test/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProviderTest.java [hunk 2]
```diff
@@ -24,22 +24,28 @@
...
+        final DataKeyAndAAD dataKeyAndAad2 = aesProvider.createDataKeyAndAAD();
 
-        assertThat(dataKey1).isNotEqualTo(dataKey2);
+        assertThat(dataKeyAndAad1).isNotEqualTo(dataKeyAndAad2);
+    }
+
+    @Test
+    void keyMustBeDifferentFromAAD() {
+        final DataKeyAndAAD dataKeyAndAad = new AesEncryptionProvider().createDataKeyAndAAD();
+        assertThat(dataKeyAndAad.dataKey.getEncoded()).isNotEqualTo(dataKeyAndAad.aad);
```
Comment (AnatolyPopov) [inline commons/src/test/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProviderTest.java:41]: Should we verify that AAD is actually there?

Comment (ivanyu) [reply commons/src/test/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProviderTest.java:49]: Yes, added the `keyAndAADMustBePresent` test.

File: commons/src/test/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProviderTest.java [hunk 3]
```diff
@@ -24,22 +24,36 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-public class EncryptionProviderTest extends RsaKeyAwareTest {
+public class AesEncryptionProviderTest extends RsaKeyAwareTest {
+
+    @Test
+    void keyAndAADMustBePresent() {
```
Comment (AnatolyPopov) [inline commons/src/test/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProviderTest.java:30]: Checkstyle

**Review by AnatolyPopov — APPROVED** [review]

**Pull request merged**